src/Services/ExportImportService.php

```diff
@@ -85,8 +106,118 @@ public function export($courseId): string
         return $zipUrl;
     }
 
-    public function import($courseId): Course
+    public function import(UploadedFile $zipFile): Model
     {
-        return new Course();
+        $dirPath = 'imports' . DIRECTORY_SEPARATOR . 'courses' . DIRECTORY_SEPARATOR . uniqid(rand(), true);
+        $dirFullPath = $this->extractZipFile($zipFile, $dirPath);
+        $content = json_decode(File::get($dirFullPath . DIRECTORY_SEPARATOR . 'content.json'), true);
+        $course = DB::transaction(function () use ($content, $dirFullPath) {
+            return $this->createCourseFromImport($content, $dirFullPath);
+        });
+
+        Storage::deleteDirectory($dirPath);
+
+        return $course;
+    }
+
+    private function extractZipFile(UploadedFile $zipFile, string $dirPath): string
+    {
+        if (Storage::exists($dirPath)) {
+            Storage::deleteDirectory($dirPath);
+        }
+
+        $dirFullPath = Storage::path($dirPath);
+        Zip::open($zipFile)->extract($dirFullPath);
+
+        return $dirFullPath;
+    }
+
+    private function createCourseFromImport(array $courseData, string $dirFullPath): Model
+    {
+        $courseData = $this->addFilesToArrayBasedOnPath($courseData, $dirFullPath);
+        $courseValidator = Validator::make($courseData, Course::$rules);
+        $course = $this->courseRepository->create($courseValidator->validate());
+
+        if (isset($courseData['lessons']) && is_array($courseData['lessons'])) {
+            foreach ($courseData['lessons'] as $lesson) {
+                $lesson['course_id'] = $course->getKey();
+                $this->createLessonFromImport($lesson, $dirFullPath);
+            }
+        }
+
+        return $course;
+    }
+
+    private function createLessonFromImport(array $lessonData, string $dirFullPath): Model
+    {
+        $lessonValidator = Validator::make($lessonData, Lesson::$rules);
+        $lesson = $this->lessonRepository->create($lessonValidator->validate());
+
+        if (isset($lessonData['topics']) && is_array($lessonData['topics'])) {
+            foreach ($lessonData['topics'] as $topic) {
+                $topic['lesson_id'] = $lesson->getKey();
+                $this->createTopicFromImport(array_filter($topic), $dirFullPath);
+            }
+        }
+
+        return $lesson;
+    }
+
+    private function createTopicFromImport(array $topicData, string $dirFullPath): Model
+    {
+        $topicData = array_merge($topicData, $topicData['topicable'] ?? []);
+        unset($topicData['topicable']);
+
+        dd(TopicFacade::availableContentClasses());
```

Comment:
'topicable_type' => ['string', Rule::in(TopicFacade::availableContentClasses())],

when I run the tests it returns empty []

when I run by postman array:7 [
  0 => "EscolaLms\TopicTypes\Models\TopicContent\Audio"
  1 => "EscolaLms\TopicTypes\Models\TopicContent\Video"
  2 => "EscolaLms\TopicTypes\Models\TopicContent\Image"
  3 => "EscolaLms\TopicTypes\Models\TopicContent\RichText"
  4 => "EscolaLms\TopicTypes\Models\TopicContent\H5P"
  5 => "EscolaLms\TopicTypes\Models\TopicContent\OEmbed"
  6 => "EscolaLms\TopicTypes\Models\TopicContent\PDF"
]

Reply:
you need to add [https://github.com/EscolaLMS/topic-types/blob/main/src/EscolaLmsTopicTypesServiceProvider.php](EscolaLmsTopicTypesServiceProvider) to testbench service providers https://github.com/EscolaLMS/Courses-Import-Export/blob/main/tests/TestCase.php#L42

you can read more about this https://packages.tools/testbench/